src/cephadm/cephadm

```diff
@@ -4232,11 +4238,25 @@ def command_run(ctx):
 ##################################
 
 
+def fsid_conf_mismatch(ctx):
+    # type: (CephadmContext) -> bool
+    (config, _) = get_config_and_keyring(ctx)
+    if config:
+        for c in config.split('\n'):
+            if 'fsid = ' in c.strip():
+                if 'fsid = ' + ctx.fsid != c.strip():
+                    return False
+    return True
+
+
 @infer_fsid
 @infer_config
 @infer_image
 def command_shell(ctx):
     # type: (CephadmContext) -> int
+    if not fsid_conf_mismatch(ctx):
+        raise Error('fsid does not match ceph conf')
```

Comment:
The boolean handling is confusing to me. "`if not ... mismatch`, then raise an error?" Can we rename this method `fsid_conf_match()`, since we raise the error when the return value is `False`?

Reply:
This is deliberate, since we don't require a match (the fsid can be omitted)... we just want to bail out if it is present AND it is a mismatch.

Reply:
Oh, the function should flip true/false around, though, you're right!